msticpy/sectools/geoip.py

```diff
@@ -278,27 +293,139 @@ class GeoLiteLookup(GeoIpLookup):
+                        print(
+                            f"Extraction complete. Local Maxmind city DB :: {_DB_FILE_PATH}"
+                        )
+            except IOError as err:
+                print(f"{_DB_ARCHIVE_PATH} {err}")
+
+    def get_geolite_dbpath(
+        self, db_folder: str = None, force_update: bool = False
+    ) -> str:
+        """ get the correct path containing GeoLite City Database
+
+        Parameters
+        ----------
+        db_folder: str, optional
+            Provide absolute path to the folder containing MMDB file (e.g. '/usr/home' or 'C:\maxmind').  
+            If no path provided, it is set to download msticpy package installation directory.
+        force_update : bool, optional
+            Force update can be set to true or false. depending on it, new download request will be initiated.
+
+        Returns
+        -------
+        str
+            Returns the absoulte path of local maxmind geolite city database after control flow logic.
+
+        """
+        if db_folder is None:
+            db_folder = self._PKG_DIR
+        force_update = self._force_update
+        _list_of_db_paths = glob.glob(db_folder + "/*.mmdb")
+
+        if _list_of_db_paths:
+            _latest_db_path = max(_list_of_db_paths, key=os.path.getmtime)
+            print(f"Local Maxmind City Database already present in {db_folder}")
+            # Check for out of date DB file with latest available
+            reader = geoip2.database.Reader(_latest_db_path)
+            # Retrive date of maxind city database accessing build_epoch property of reader object.
+            _last_mod_time = datetime.utcfromtimestamp(reader.metadata().build_epoch)
+            _db_age = datetime.utcnow() - _last_mod_time
+            if _db_age > timedelta(30):
+                print(
+                    f"Latest local Maxmind City Database present is older than 30 days. Downloading new database to {db_folder}"
+                )
+                self.download_and_extract_gzip(self._MAXMIND_DOWNLOAD, db_folder)
+            else:
+                if force_update:
+                    print(
+                        f"Force update is set to True. Downloading new database to {db_folder}"
+                    )
+                    self.download_and_extract_gzip(self._MAXMIND_DOWNLOAD, db_folder)
+                else:
+                    print(
+                        "Existing local Maxmind City Database is downloaded in last 30 days so it will be used"
+                    )
+        else:
+            print(
+                f"No local Maxmind City Database found. Downloading new database to {db_folder}"
+            )
+            self.download_and_extract_gzip(self._MAXMIND_DOWNLOAD, db_folder)
+            _list_of_db_paths = glob.glob(db_folder + "/*.mmdb")
+            _latest_db_path = _list_of_db_paths[0]
+
+        return _latest_db_path
+
+    def __init__(self, db_folder: str = None, force_update: bool = False):
```

Comment:
Keep as top method in class.
Add "auto_update" parameter to allow people to disable auto-update

Reply:
added option to toggle auto_update